packages/synthetics-sdk-broken-links/src/broken_links.ts

```diff
@@ -33,39 +33,97 @@ import {
 export async function runBrokenLinks(
   options: BrokenLinksResultV1_BrokenLinkCheckerOptions
 ): Promise<SyntheticResult> {
-  // START - to resolve warnings while under development
-  checkStatusPassing({ status_value: 200 } as ResponseStatusCode, 200);
-
-  const browser = await puppeteer.launch({ headless: 'new' });
-  const page = await browser.newPage();
-  await checkLink(
-    page,
-    {} as LinkIntermediate,
-    {} as BrokenLinksResultV1_BrokenLinkCheckerOptions
-  );
-  // END - to resolve warnings  while under development
-
   // options object modified directly
   setDefaultOptions(options);
 
-  // PSEUDOCODE
-
-  // create puppeteer.Browser
-  // create puppeteer.Page & navigate to origin_url, w/ origin specific settings
+  // create Browser & origin page then navigate to origin_url, w/ origin
+  // specific settings
+  const browser = await puppeteer.launch({ headless: 'new' });
+  const origin_page = await browser.newPage();
+  if (options.wait_for_selector)
+    await origin_page.waitForSelector(options.wait_for_selector);
 
   // scrape origin_url for all links
-  // (shuffle links if necessary and) truncate at link_limit
+  const retrieved_links: LinkIntermediate[] = await retrieveLinksFromPage(
+    origin_page,
+    options.query_selector_all,
+    options.get_attributes
+  );
+
+  // TODO:  (shuffle links if necessary and) truncate at link_limit
 
+  // TODO
   // create new page to be used for all scraped links
   // navigate to each link - LOOP:
   //          each call to `checkLinks(...)` will return a `SyntheticLinkResult`
   //          Object added to an array of `followed_links`
 
   // returned a SyntheticResult with `options`, `followed_links` &
   // runtimeMetadata
+
+  // Resolving warnings - START IGNORE (will be removed)
+  await checkLink(
+    origin_page,
+    {} as LinkIntermediate,
```

Comment:
doing {} as LinkIntermediate is actually not a good practice because that interface doesnt have optional fields. IE target_url is not optional.

I think that what you should do is either...
1) Create actual instances (albeit empty) of LinkIntermediate and BrokenLinksResultV1_BrokenLinkCheckerOptions
2) Overload checkLink to take just an origin_page OR origin_page + params

Reply:
this is just to resolve the checkLink warning. in the future once the function has been fully implemented it will not be called with casting.

To avoid a situation where I am sending out PRs with helpers that are not actually called I wrote pseudocode in `runBrokenLinks()` and `checkLink()` so that reviewers would have context for how these helper functions are being called. 
Overall, there will be no casting in the future, lines 103 - 110 are just to resolve warnings